File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/gameLogs/types/UnitModel.ts
import type { UnitHealthModel } from './UnitHealthModel';
import type { UnitCastModel } from './UnitCastModel';
import type { UnitPositionModel } from './UnitPositionModel';
import type { UnitInfoModel } from './UnitInfoModel';

export type UnitModel = {
    id: string;
    gameId: string;
    name: string;
    unitHash: string;
    type: number;
    creatorGameId: string | null;
    unitInfo: UnitInfoModel;
    unitHealthes: UnitHealthModel[];
    unitCasts: UnitCastModel[];
    unitPositions: UnitPositionModel[];
    combatId: number;
}

File: src/Web/CombatAnalysis.EnhancedWebApp/CombatAnalysis.EnhancedWebApp.Server/Models/GameLogs/UnitModel.cs
namespace CombatAnalysis.EnhancedWebApp.Server.Models.GameLogs;

public class UnitModel
{
    public string Id { get; set; } = string.Empty;

    public string GameId { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string UnitHash { get; set; } = string.Empty;

    public int Type { get; set; }

    public string? CreatorGameId { get; set; }
    
    public int CombatId { get; set; }

    public UnitInfoModel UnitInfo { get; set; }

    public List<UnitHealthModel> UnitHealthes { get; set; } = [];

    public List<UnitCastModel> UnitCasts { get; init; } = [];

    public List<UnitPositionModel> UnitPositions { get; set; } = [];
}
